docs/rfcs/2025-11-17-async-commands-acks.md

```diff
@@ -0,0 +1,74 @@
+# Acknowledgments for async command responses
+
+## Problem
+
+Continuations for asynchronous commands can be forever lost if their execution fails, e.g. due to a crash. This can result in failures in establishing connections, sending post-connection auto-reply, etc. depending on other applications of asynchronous commands.
+
+## Solution
+
+An idea is to persist events in agent until chat acknowledges their processing, and replay them on next start of command processing.
+
+### Agent persistence
+
+Save response on command before notifying chat (event received by chat via subQ).
+
+```sql
+ALTER TABLE commands ADD COLUMN response BLOB;
+```
+
+Type is `AEvent`, requires encoding for To and FromField instances.
+
+Application of chat continuations is very limited, so not all events need to be saved. In fact currently we only need 2 types of events to be recorded - see below. This breaks separation between chat and agent (agent knows which events to record), however that abstraction has long been violated. This can be a contract between agent and chat - which events to keep and acknowledge.
+
+TBC separate type for storing only necessary events:
+
+```haskell
+data AEventDB where
```

Comment:
even if we want to persist some commands, no need to define type - it can be just a Bool predicate.

Reply:
this is just to consider if we want to do less encodings